From a78f431d8ec01aa69ccc860be598d65f7bfb7826 Mon Sep 17 00:00:00 2001
Message-ID: <a78f431d8ec01aa69ccc860be598d65f7bfb7826.1752540533.git.sam@gentoo.org>
In-Reply-To: <a13efe50cc12de7874d1106207fef600bce9aada.1752540533.git.sam@gentoo.org>
References: <a13efe50cc12de7874d1106207fef600bce9aada.1752540533.git.sam@gentoo.org>
From: Sam James <sam@gentoo.org>
Date: Thu, 3 Jul 2025 13:27:13 +0100
Subject: [PATCH 2/2] gas: disable SFrames for some tests

These break with a gas configured to emit SFrames by default.

	* testsuite/gas/cfi/cfi-label.d: Pass --gsframe=no.
	* testsuite/gas/cfi/cfi-x86_64.d: Ditto.
	* testsuite/gas/scfi/x86_64/scfi-cfi-label-1.d: Ditto.
	* testsuite/gas/scfi/x86_64/scfi-x86-64.exp: Ditto.
---
 gas/testsuite/gas/cfi/cfi-label.d                | 2 +-
 gas/testsuite/gas/cfi/cfi-x86_64.d               | 2 +-
 gas/testsuite/gas/scfi/x86_64/scfi-cfi-label-1.d | 2 +-
 gas/testsuite/gas/scfi/x86_64/scfi-x86-64.exp    | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/gas/testsuite/gas/cfi/cfi-label.d b/gas/testsuite/gas/cfi/cfi-label.d
index c6507138b43..07bc3944965 100644
--- a/gas/testsuite/gas/cfi/cfi-label.d
+++ b/gas/testsuite/gas/cfi/cfi-label.d
@@ -1,4 +1,4 @@
-#as: -mx86-used-note=no --generate-missing-build-notes=no
+#as: -mx86-used-note=no --generate-missing-build-notes=no --gsframe=no
 #objdump: -tWf
 #name: .cfi_label directive
 
diff --git a/gas/testsuite/gas/cfi/cfi-x86_64.d b/gas/testsuite/gas/cfi/cfi-x86_64.d
index 6dd58f50cf5..29236cd9584 100644
--- a/gas/testsuite/gas/cfi/cfi-x86_64.d
+++ b/gas/testsuite/gas/cfi/cfi-x86_64.d
@@ -1,4 +1,4 @@
-#as: -O0
+#as: -O0 --gsframe=no
 #objdump: -Wf
 #name: CFI on x86-64
 #...
diff --git a/gas/testsuite/gas/scfi/x86_64/scfi-cfi-label-1.d b/gas/testsuite/gas/scfi/x86_64/scfi-cfi-label-1.d
index 7a1d1715262..8e390152f7d 100644
--- a/gas/testsuite/gas/scfi/x86_64/scfi-cfi-label-1.d
+++ b/gas/testsuite/gas/scfi/x86_64/scfi-cfi-label-1.d
@@ -1,4 +1,4 @@
-#as: --scfi=experimental -W
+#as: --scfi=experimental -W --gsframe=no
 #as:
 #objdump: -tWf
 #name: SCFI no ignore .cfi_label
diff --git a/gas/testsuite/gas/scfi/x86_64/scfi-x86-64.exp b/gas/testsuite/gas/scfi/x86_64/scfi-x86-64.exp
index 880948dd89c..4bba2a78007 100644
--- a/gas/testsuite/gas/scfi/x86_64/scfi-x86-64.exp
+++ b/gas/testsuite/gas/scfi/x86_64/scfi-x86-64.exp
@@ -48,7 +48,7 @@ if  { ([istarget "x86_64-*-*"] && ![istarget "x86_64-*-linux*-gnux32"]) } then {
     run_list_test "ginsn-cofi-1" "--scfi=experimental -ali -W"
 
     run_dump_test "scfi-cfi-label-1"
-    run_list_test "scfi-cfi-label-1" "--scfi=experimental --warn"
+    run_list_test "scfi-cfi-label-1" "--scfi=experimental --warn --gsframe=no"
     run_dump_test "scfi-cfi-sections-1"
     run_list_test "scfi-cfi-sections-1" "--scfi=experimental --warn"
 
-- 
2.50.1

